### ISSUE TITLE
device_properties_t.max_frequency always returns 0

### ISSUE THREAD

**Issue by Withaust** [ThePhD/infoware#65]
It happens for both discrete and integrated video cards with DX detection for some reason.

**Comment by nabijaczleweli** [comment]
Yes: https://github.com/ThePhD/infoware/blob/29538ab30c402cc105eafef427dcd9a329d6aeb2/src/gpu/d3d.cpp#L60-L63

This is available through OpenCL only at present. I don't immediately see a way to see this documented in [the D3D API](https://learn.microsoft.com/en-us/windows/win32/api/dxgi/ns-dxgi-dxgi_adapter_desc) either.
